src/Microsoft.VisualStudio.Editor.Razor/DefaultTagHelperResolver.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal class DefaultTagHelperResolver : TagHelperResolver
+    {
+        private readonly ErrorReporter _errorReporter;
+
+        // Hack for testability. The view component visitor will normally just no op if we're not referencing
+        // an appropriate version of MVC.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        internal bool ForceEnableViewComponentDiscovery { get; set; }
+
+        public DefaultTagHelperResolver(ErrorReporter errorReporter)
+        {
+            if (errorReporter == null)
+            {
+                throw new ArgumentNullException(nameof(errorReporter));
+            }
+
+            _errorReporter = errorReporter;
+        }
+
+        public override async Task<TagHelperResolutionResult> GetTagHelpersAsync(Project project, CancellationToken cancellationToken)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            try
+            {
+                var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var result = GetTagHelpers(compilation);
+                return result;
+            }
+            catch (Exception exception)
+            {
+                _errorReporter.ReportError(exception, project);
```

Comment:
I don't think we need to report this exception here, it's going to bubble. Whomever is the caller should be handling this and logging it.

Reply:
I'm fine with not having any throw cases in the editor.razor variant. However, in the windows resolver going to remove the error reporter log but keep the try/catch => throw unexpected error since we're calling into Roslyn bits that deal with calling oop.